src/core/ext/filters/client_channel/http_proxy.c

```diff
@@ -56,8 +58,6 @@ static char* get_http_proxy_server(grpc_exec_ctx* exec_ctx, char** user_cred) {
     goto done;
   }
   /* Split on '@' to separate user credentials from host */
-  char** authority_strs = NULL;
```

Comment:
Why move these declarations?  I prefer to declare variables right before they are used, since IMHO this makes the code more readable.

Reply:
We cross declarations with gotos. This is not allowed in C++. To do otherwise would mean significant refactoring, and since that'll happen anyway (in a parallelizable way) once we're compiling as c++, I'd hold that we should prefer the most mechanical transform possible at this point.

Reply:
Blech.  Okay, I didn't realize this was actually a C++ compilation problem.

Hopefully, once we actually start using C++ features, we can eliminate the gotos by using the equivalent of `unique_ptr<>` to automatically clean things up when the function exits.